source/kraken/worker.cpp

```diff
@@ -772,6 +752,31 @@ void Worker::journeys(const pbnavitia::JourneysRequest& request, pbnavitia::API
     }
 }
 
+static std::pair<const type::EntryPoint, const boost::optional<const type::EntryPoints&>> get_center_and_stop_points(
+    const JourneysArg& arg) {
+    // If we have a center
+    // Origins or destinations are already computed stop_points
+    const auto& center =
+        arg.isochrone_center ? *arg.isochrone_center : (arg.origins.empty() ? arg.destinations[0] : arg.origins[0]);
```

Comment:
Memory safety: Do you make sure somewhere that `args.destinations` isn't empty if you use it?
Also, better to always use `.at(0)` to fail loudly :wink:

Reply:
Also, if you keep the same behavior (not :avocado:-ing for this at all), you may fill `stop_points` first, then `center` like this:
```suggestion
        arg.isochrone_center ? *arg.isochrone_center : stop_points.at(0);
```

Reply:
> Memory safety: Do you make sure somewhere that `args.destinations` isn't empty if you use it?
> Also, better to always use `.at(0)` to fail loudly wink

Yes, it is done in set_journeys_args. But I can use `at(0)` or `front()`

Reply:
Yes, it is done in set_journeys_args. But I can use `at(0)` or `front()`

Reply:
Your suggestion wouldn't work. If there is no isochrone_center, stop_points is `boost::none`.

Reply:
I've added some `front()`